### PR TITLE
audio-sharing: init at 0.2.2

### PR DESCRIPTION
**Pull request by BenediktBroich**
## Description of changes

Application for sharing audio in you network (rtsp). 

## Test

1. `audio-share`
2. copy link
3. `mpv <LINK>` now you should hear the audio

Src: [AudioSharing](https://gitlab.gnome.org/World/AudioSharing)
Issue: https://github.com/NixOS/nixpkgs/issues/195280

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/audio-sharing/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "audio-sharing";
+  version = "0.2.2";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = "AudioSharing";
+    rev = "v${finalAttrs.version}";
```
Comment (linsui) [inline pkgs/applications/audio/audio-sharing/default.nix:28]:
```suggestion
    rev = finalAttrs.version;
```

**Comment by BenediktBroich** [comment]
Did you use 

> It still doesn't work for me. It says No audio sink found.

True. Dependency pulsesrc was missing. Now you can test again if you want to.

**Comment by linsui** [comment]
Still the same error.

**Comment by BenediktBroich** [comment]
Result of `nixpkgs-review pr 259976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>audio-sharing</li>
  </ul>
</details>

**Comment by BenediktBroich** [comment]
@linsui What sound config do you have on your system?

 `  sound = {
    enable = true;
  };
`

**Comment by linsui** [comment]
```nix
    pipewire = {
      enable = true;
      alsa.enable = true;
      alsa.support32Bit = true;
      pulse.enable = true;
    };
```

**Comment by BenediktBroich** [comment]
@linsui Pipewire does not work see  [ISSUE](https://gitlab.gnome.org/World/AudioSharing/-/issues/16).

**Comment by BenediktBroich** [comment]
> @linsui What sound config do you have on your system?
> 
>

**Comment by linsui** [comment]
Ah, that's sad... Yes, after switching to pipewire audio sharing can't find sink anymore. Though it didn't work for me with pulseaudio, either.

**Comment by BenediktBroich** [comment]
@linsui Did you build it using nix-build or nixpkgs-review?

**Comment by linsui** [comment]
I build it with nix-build.

**Comment by BenediktBroich** [comment]
Can you run `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 259976"` inside of you nixpkgs folder?

**Comment by linsui** [comment]
Nothing changed. By the way, nixpkgs-review is really slow...

**Comment by BenediktBroich** [comment]
Can you post the nix-build output?

**Comment by linsui** [comment]
The log or the file?

**Comment by BenediktBroich** [comment]
Just the terminal output of the `nix-build $NIXPKGS -A audio-sharing --check` command.

**Comment by linsui** [comment]
[out.log](https://github.com/NixOS/nixpkgs/files/12848726/out.log)

**Comment by BenediktBroich** [comment]
I don't see Issue. Maybe somone else can test it first, to confirm this.

**Comment by auroraanna** [comment]
The test works for me.

**Comment by gaykitty** [comment]
Result of `nixpkgs-review pr 259976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>audio-sharing</li>
  </ul>
</details>

**Comment by gaykitty** [comment]
I ran `nixpkgs-review pr 259976`. Everything built fine but when I run it I get "No audio sink found". 

Here's my audio config and the build log:

```nix
  sound.enable = true;
  hardware.pulseaudio.enable = false;
  security.rtkit.enable = true;
  services.pipewire = {
    enable = true;
    alsa.enable = true;
    alsa.support32Bit = true;
    pulse.enable = true;
    jack.enable = true;
  };
```

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/5y0yf46hkp5aib08plljzyjy52wkbkgx-source
source root is source
Executing cargoSetupPostUnpackHook
unpacking source archive /nix/store/l3nfdf68xdcrarabd5ah8f8nf3sg1as1-audio-sharing-0.2.2-vendor.tar.gz
Finished cargoSetupPostUnpackHook
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
Executing cargoSetupPostPatchHook
Validating consistency between /build/source/Cargo.lock and /build/audio-sharing-0.2.2-vendor.tar.gz/Cargo.lock
Finished cargoSetupPostPatchHook
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
mesonConfigurePhase flags: --prefix=/nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2 --libdir=/nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/lib --libexecdir=/nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/libexec --bindir=/nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/bin --sbindir=/nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/sbin --includedir=/nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/include --mandir=/nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/man --infodir=/nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/info --localedir=/nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale -Dauto_features=enabled -Dwrap_mode=nodownload --buildtype=plain
[1mThe Meson build system[0m
Version: 1.2.3
Source dir: [1m/build/source[0m
Build dir: [1m/build/source/build[0m
Build type: [1mnative build[0m
Project name: [1maudio-sharing[0m
Project version: [1m0.2.2[0m
Rust compiler for the host machine: [1mrustc -C linker=gcc[0m (rustc 1.74.0)
Rust linker for the host machine: [1mrustc -C linker=gcc[0m ld.bfd 2.40
Host machine cpu family: [1mx86_64[0m
Host machine cpu: [1mx86_64[0m
Found pkg-config: [1m/nix/store/1nvm1a9a9mlwyy7d0hapkzchxgjgpw19-pkg-config-wrapper-0.29.2/bin/pkg-config[0m (0.29.2)
Run-time dependency [1mdbus-1[0m found: [1;32mYES[0m [36m1.14.10[0m
Run-time dependency [1mglib-2.0[0m found: [1;32mYES[0m [36m2.78.1[0m
Run-time dependency [1mgio-2.0[0m found: [1;32mYES[0m [36m2.78.1[0m
Run-time dependency [1mgtk4[0m found: [1;32mYES[0m [36m4.12.3[0m
Run-time dependency [1mlibadwaita-1[0m found: [1;32mYES[0m [36m1.4.2[0m
Run-time dependency [1mgstreamer-1.0[0m found: [1;32mYES[0m [36m1.22.7[0m
Run-time dependency [1mgstreamer-base-1.0[0m found: [1;32mYES[0m [36m1.22.7[0m
Run-time dependency [1mgstreamer-audio-1.0[0m found: [1;32mYES[0m [36m1.22.7[0m
Run-time dependency [1mgstreamer-plugins-base-1.0[0m found: [1;32mYES[0m [36m1.22.7[0m
Run-time dependency [1mgstreamer-rtsp-server-1.0[0m found: [1;32mYES[0m [36m1.22.7[0m
[1;33mWARNING:[0m You should add the boolean check kwarg to the run_command call.
         It currently defaults to false,
         but it will default to true in future releases of meson.
         See also: https://github.com/mesonbuild/meson/issues/9300
Configuring [1mde.haeckerfelix.AudioSharing.desktop.in[0m using configuration
Program [1mmsgfmt[0m found: [1;32mYES[0m (/nix/store/f0m95a8gfkwgijbiscbkn7f3ra6ha70x-gettext-0.21.1/bin/msgfmt)
Program [1mdesktop-file-validate[0m found: [1;32mYES[0m (/nix/store/mqjqydhpimcxcf7vz3bjnx54fz7ypma0-desktop-file-utils-0.27/bin/desktop-file-validate)
Configuring [1mde.haeckerfelix.AudioSharing.metainfo.xml.in[0m using configuration
Program [1mappstream-util[0m found: [1;32mYES[0m (/nix/store/zkag0x9zv6pr2sz88b54yvl7hr8zw7gr-appstream-glib-0.8.2-dev/bin/appstream-util)
Configuring [1mde.haeckerfelix.AudioSharing.service[0m using configuration
Configuring [1mde.haeckerfelix.AudioSharing.gresource.xml[0m using configuration
Found pkg-config: [1m/nix/store/1nvm1a9a9mlwyy7d0hapkzchxgjgpw19-pkg-config-wrapper-0.29.2/bin/pkg-config[0m (0.29.2)
Dependency [1mgio-2.0[0m found: [1;32mYES[0m [36m2.78.1[0m [1;34m(cached)[0m
Program [1m/nix/store/xc5casr35bpf24dj0sp3jjwh76mkslg4-glib-2.78.1-dev/bin/glib-compile-resources[0m found: [1;32mYES[0m (/nix/store/xc5casr35bpf24dj0sp3jjwh76mkslg4-glib-2.78.1-dev/bin/glib-compile-resources)
Program [1mgtk4-update-icon-cache[0m found: [1;32mYES[0m (/nix/store/2x3zx0jyna3isnhvbq2s9xlhdn4jhdz0-gtk4-4.12.3/bin/gtk4-update-icon-cache)
Program [1mupdate-desktop-database[0m found: [1;32mYES[0m (/nix/store/mqjqydhpimcxcf7vz3bjnx54fz7ypma0-desktop-file-utils-0.27/bin/update-desktop-database)
Configuring [1mconfig.rs[0m using configuration
Program [1mcargo[0m found: [1;32mYES[0m (/nix/store/czzzpyqqvd62yxmdnhhil6jz732s8q0w-cargo-1.74.0/bin/cargo)
[1mMessage:[0m Building in release mode
Program [1mmsginit[0m found: [1;32mYES[0m (/nix/store/f0m95a8gfkwgijbiscbkn7f3ra6ha70x-gettext-0.21.1/bin/msginit)
Program [1mmsgmerge[0m found: [1;32mYES[0m (/nix/store/f0m95a8gfkwgijbiscbkn7f3ra6ha70x-gettext-0.21.1/bin/msgmerge)
Program [1mxgettext[0m found: [1;32mYES[0m (/nix/store/f0m95a8gfkwgijbiscbkn7f3ra6ha70x-gettext-0.21.1/bin/xgettext)
Build targets in project: [1m36[0m

audio-sharing [36m0.2.2[0m

  [1mUser defined options[0m
    auto_features: enabled
    bindir       : /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/bin
    buildtype    : plain
    includedir   : /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/include
    infodir      : /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/info
    libdir       : /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/lib
    libexecdir   : /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/libexec
    localedir    : /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale
    mandir       : /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/man
    prefix       : /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2
    sbindir      : /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/sbin
    wrap_mode    : nodownload

Found ninja-1.11.1 at /nix/store/7p8kahc23klw4lfx1d7i5gs2547mbcj5-ninja-1.11.1/bin/ninja
mesonConfigurePhase: enabled\ parallel\ building
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
build flags: -j16
[0/33] Generating src/cargo-build with a custom command
   Compiling autocfg v1.1.0
   Compiling serde v1.0.163
   Compiling hashbrown v0.12.3
   Compiling winnow v0.4.6
   Compiling target-lexicon v0.12.7
   Compiling heck v0.4.1
   Compiling smallvec v1.10.0
   Compiling version-compare v0.1.1
   Compiling pkg-config v0.3.27
   Compiling libc v0.2.144
   Compiling proc-macro2 v1.0.59
   Compiling unicode-ident v1.0.9
   Compiling quote v1.0.28
   Compiling syn v1.0.109
   Compiling version_check v0.9.4
   Compiling memchr v2.5.0
   Compiling futures-core v0.3.28
   Compiling futures-task v0.3.28
   Compiling anyhow v1.0.71
   Compiling futures-util v0.3.28
   Compiling once_cell v1.17.1
   Compiling thiserror v1.0.40
   Compiling proc-macro-error-attr v1.0.4
   Compiling proc-macro-error v1.0.4
   Compiling indexmap v1.9.3
   Compiling slab v0.4.8
   Compiling futures-channel v0.3.28
   Compiling pin-project-lite v0.2.9
   Compiling pin-utils v0.1.0
   Compiling bitflags v1.3.2
   Compiling num-traits v0.2.15
   Compiling num-integer v0.1.45
   Compiling gio v0.15.12
   Compiling futures-io v0.3.28
   Compiling semver v1.0.17
   Compiling paste v1.0.12
   Compiling cfg-expr v0.15.1
   Compiling num-rational v0.4.1
   Compiling memoffset v0.8.0
   Compiling muldiv v1.0.1
   Compiling syn v2.0.18
   Compiling cfg-if v1.0.0
   Compiling aho-corasick v1.0.1
   Compiling pnet_base v0.28.0
   Compiling pretty-hex v0.3.0
   Compiling regex-syntax v0.7.2
   Compiling rustc_version v0.4.0
   Compiling num-iter v0.1.43
   Compiling num-rational v0.3.2
   Compiling field-offset v0.3.5
   Compiling log v0.4.18
   Compiling temp-dir v0.1.11
   Compiling cc v1.0.79
   Compiling option-operations v0.4.1
   Compiling pnet_macros_support v0.28.0
   Compiling quick-xml v0.22.0
   Compiling pnet_sys v0.28.0
   Compiling quick-error v1.2.3
   Compiling humantime v1.3.0
   Compiling futures-executor v0.3.28
   Compiling atty v0.2.14
   Compiling bytemuck v1.13.1
   Compiling color_quant v1.1.0
   Compiling lazy_static v1.4.0
   Compiling byteorder v1.4.3
   Compiling pnet v0.28.0
   Compiling termcolor v1.2.0
   Compiling regex v1.8.3
   Compiling gettext-sys v0.21.3
   Compiling image v0.23.14
   Compiling checked_int_cast v1.0.0
   Compiling gtk-macros v0.3.0
   Compiling thiserror-impl v1.0.40
   Compiling locale_config v0.3.0
   Compiling toml_datetime v0.6.2
   Compiling serde_spanned v0.6.2
   Compiling env_logger v0.7.1
   Compiling gettext-rs v0.7.0
   Compiling toml_edit v0.19.10
   Compiling pretty_env_logger v0.4.0
   Compiling ipnetwork v0.18.0
   Compiling qrcode v0.12.0
   Compiling pnet_macros v0.28.0
   Compiling pnet_datalink v0.28.0
   Compiling toml v0.7.4
   Compiling proc-macro-crate v1.3.1
   Compiling system-deps v6.1.0
   Compiling pnet_packet v0.28.0
   Compiling glib-sys v0.15.10
   Compiling gobject-sys v0.15.10
   Compiling gio-sys v0.15.10
   Compiling glib-macros v0.15.13
   Compiling pango-sys v0.15.10
   Compiling cairo-sys-rs v0.15.1
   Compiling gstreamer-sys v0.18.0
   Compiling gdk-pixbuf-sys v0.15.10
   Compiling gdk4-sys v0.4.8
   Compiling graphene-sys v0.15.10
   Compiling gsk4-sys v0.4.8
   Compiling gstreamer-sdp-sys v0.18.0
   Compiling gtk4-sys v0.4.8
   Compiling gstreamer-rtsp-sys v0.18.0
   Compiling gstreamer-net-sys v0.18.0
   Compiling libadwaita-sys v0.2.0-alpha.3
   Compiling gstreamer-rtsp-server-sys v0.18.0
   Compiling gtk4-macros v0.4.10
   Compiling pnet_transport v0.28.0
   Compiling glib v0.15.12
   Compiling gstreamer v0.18.8
   Compiling cairo-rs v0.15.12
   Compiling pango v0.15.10
   Compiling graphene-rs v0.15.1
   Compiling gstreamer-sdp v0.18.0
   Compiling gstreamer-rtsp v0.18.0
   Compiling gdk-pixbuf v0.15.11
   Compiling gstreamer-net v0.18.0
   Compiling gstreamer-rtsp-server v0.18.7
   Compiling gdk4 v0.4.8
   Compiling gsk4 v0.4.8
   Compiling gtk4 v0.4.9
   Compiling libadwaita v0.2.0-alpha.3
   Compiling audio-sharing v0.0.0 (/build/source)
    Finished release [optimized] target(s) in 1m 13s
warning: the following packages contain code that will be rejected by a future version of Rust: quick-xml v0.22.0
note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 1`
[13/33] Generating data/de.haeckerfelix.AudioSharing_gresource with a custom command
xml-stripblanks preprocessing requested, but XMLLINT is not set, and xmllint is not in PATH
[32/33] Generating data/de.haeckerfelix.AudioSharing.metainfo.xml with a custom command
buildPhase completed in 1 minutes 14 seconds
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Running phase: installPhase
mesonInstallPhase flags: ''
Installing data/de.haeckerfelix.AudioSharing.desktop to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/applications
Installing data/de.haeckerfelix.AudioSharing.metainfo.xml to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/metainfo
Installing data/de.haeckerfelix.AudioSharing.gresource to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/audio-sharing
Installing src/audio-sharing to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/bin
Installing po/be/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/be/LC_MESSAGES
Installing po/da/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/da/LC_MESSAGES
Installing po/de/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/de/LC_MESSAGES
Installing po/es/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/es/LC_MESSAGES
Installing po/eu/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/eu/LC_MESSAGES
Installing po/fa/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/fa/LC_MESSAGES
Installing po/fi/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/fi/LC_MESSAGES
Installing po/fr/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/fr/LC_MESSAGES
Installing po/fur/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/fur/LC_MESSAGES
Installing po/gl/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/gl/LC_MESSAGES
Installing po/he/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/he/LC_MESSAGES
Installing po/hi/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/hi/LC_MESSAGES
Installing po/hr/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/hr/LC_MESSAGES
Installing po/hu/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/hu/LC_MESSAGES
Installing po/id/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/id/LC_MESSAGES
Installing po/is/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/is/LC_MESSAGES
Installing po/it/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/it/LC_MESSAGES
Installing po/ka/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/ka/LC_MESSAGES
Installing po/lt/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/lt/LC_MESSAGES
Installing po/nl/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/nl/LC_MESSAGES
Installing po/oc/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/oc/LC_MESSAGES
Installing po/pt/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/pt/LC_MESSAGES
Installing po/pt_BR/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/pt_BR/LC_MESSAGES
Installing po/ru/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/ru/LC_MESSAGES
Installing po/sl/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/sl/LC_MESSAGES
Installing po/sr/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/sr/LC_MESSAGES
Installing po/sv/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/sv/LC_MESSAGES
Installing po/tr/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/tr/LC_MESSAGES
Installing po/uk/LC_MESSAGES/audio-sharing.mo to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/locale/uk/LC_MESSAGES
Installing /build/source/data/icons/hicolor/scalable/apps/de.haeckerfelix.AudioSharing.svg to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/icons/hicolor/scalable/apps
Installing /build/source/data/icons/hicolor/symbolic/apps/de.haeckerfelix.AudioSharing-symbolic.svg to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/icons/hicolor/symbolic/apps
Installing /build/source/build/data/de.haeckerfelix.AudioSharing.service to /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/dbus-1/services
Running custom install script '/nix/store/2x3zx0jyna3isnhvbq2s9xlhdn4jhdz0-gtk4-4.12.3/bin/gtk4-update-icon-cache -q -t -f /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/icons/hicolor'
Running custom install script '/nix/store/mqjqydhpimcxcf7vz3bjnx54fz7ypma0-desktop-file-utils-0.27/bin/update-desktop-database -q /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/applications'
@nix { "action": "setPhase", "phase": "glibPreFixupPhase" }
Running phase: glibPreFixupPhase
@nix { "action": "setPhase", "phase": "mimeinfoPreFixupPhase" }
Running phase: mimeinfoPreFixupPhase
@nix { "action": "setPhase", "phase": "glibPreFixupPhase" }
Running phase: glibPreFixupPhase
@nix { "action": "setPhase", "phase": "glibPreFixupPhase" }
Running phase: glibPreFixupPhase
@nix { "action": "setPhase", "phase": "glibPreFixupPhase" }
Running phase: glibPreFixupPhase
@nix { "action": "setPhase", "phase": "gappsWrapperArgsHook" }
Running phase: gappsWrapperArgsHook
@nix { "action": "setPhase", "phase": "dropIconThemeCache" }
Running phase: dropIconThemeCache
Removing /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/share/icons/hicolor/icon-theme.cache
@nix { "action": "setPhase", "phase": "dropIconThemeCache" }
Running phase: dropIconThemeCache
@nix { "action": "setPhase", "phase": "dropIconThemeCache" }
Running phase: dropIconThemeCache
@nix { "action": "setPhase", "phase": "fixupPhase" }
Running phase: fixupPhase
Wrapping program '/nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/bin/audio-sharing'
shrinking RPATHs of ELF executables and libraries in /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2
shrinking /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/bin/audio-sharing
shrinking /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/bin/.audio-sharing-wrapped
checking for references to /build/ in /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2...
patching script interpreter paths in /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2
stripping (with command strip and flags -S -p) in  /nix/store/agfng7g9hlsd6m1ra3p5vfzjkbmw6g08-audio-sharing-0.2.2/bin
```

**Comment by gaykitty** [comment]
Also note, the flatpak version of this app does not have this issue on my system.

**Comment by auroraanna** [comment]
@gaykitty Why are you enabling ALSA (`sound`) when you (I assume) want to be using pipewire?

**Comment by auroraanna** [comment]
I seems that it uses gstreamer for getting the audio but I failed to find out which audio servers it supports.

**Comment by auroraanna** [comment]
dafuq… I tried to find out which sound servers you need by enabling and disabling the pipewire plugins but I found out that you need both the alsa and pulseaudio emulation.

**Comment by gaykitty** [comment]
> @gaykitty Why are you enabling ALSA (`sound`) when you (I assume) want to be using pipewire?

tbh idk. That part of my config has been there since I first started using nixos. I think I just copied it from someone.

**Comment by auroraanna** [comment]
@gaykitty does it work without?

**Comment by gaykitty** [comment]
Removing `sound.enable = true;` does not fix it.

**Comment by auroraanna** [comment]
@gaykitty so you have the same audio config as me but it doesn't work? Confused…

**Comment by gaykitty** [comment]
If you're config is:

```nix
hardware.pulseaudio.enable = false;
security.rtkit.enable = true;
services.pipewire = {
  enable = true;
  alsa.enable = true;
  alsa.support32Bit = true
  pulse.enable = true;
  jack.enable = true;
};
```
Then yes.

**Comment by auroraanna** [comment]
yes, that's it.

**Comment by auroraanna** [comment]
maybe some other config options have an effect on this?

**Comment by BenediktBroich** [comment]
Result of `nixpkgs-review pr 259976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>audio-sharing</li>
  </ul>
</details>

**Comment by amaxine** [comment]
Works for me with a similar config to https://github.com/NixOS/nixpkgs/pull/259976#issuecomment-1889116521 (without jack). I'm not sure I like merging it with a very unclear reason for why it does or doesn't work for different people, seeing as there's complaints on both pulse and pipewire, but not enough for me to take a guess at a repro case and debugging myself.

@BenediktBroich regardless of the above, this should be following the by-name scheme now before it can be merged.

**Comment by BenediktBroich** [comment]
The application is using gst_all_1. And it is only supposed to work with pulseaudio ([here](https://gitlab.gnome.org/search?search=pulse&nav_source=navbar&project_id=16040&group_id=362&search_code=true&repository_ref=main)). It will only work with pipewire when backwards the `compatiblity services.pipewire.pulse.enable = true;` or `hardware.pulseaudio.enable = enable;`. I don't know if i am supposed to check these dependencies in this package.nix. 

So i would suppose to merge this Package and i will have a lookout for Issues tagging audio-share to further debug this. Otherwise i think we will not get much feedback on this.

**Comment by amaxine** [comment]
Good point.

**Pull request merged**